sdcm/remote.py

```diff
@@ -963,3 +967,126 @@ def run_output_check(self, cmd, timeout=None, ignore_status=False,
+
+    # def __init__(self, hostname, user="root", port=22, password="",
+                 # key_file=None, wait_key_installed=0, extra_ssh_options=""):
+    def __init__(self, *args, **kwargs):
+        logger = logging.getLogger('avocado.test')
+        self.log = SDCMAdapter(logger, extra={'prefix': str(self)})
+        self.log.info(args)
+        self.log.info(kwargs)
+        self.key_file = kwargs.get('key_file', "")
+        self.user = kwargs.get('user', "scylla-test")
+        self.port = kwargs.get('port', 22)
+        self.hostname = kwargs.get('hostname', "localhost")
+        self.password = kwargs.get('password', "")
+        self.connect_timeout = kwargs.get('connect_timeout', 60)
+        self.cmd_timeout = kwargs.get('timeout', 300)
+        self.known_hosts_file = tempfile.mkstemp()[1]
+        config = Config(overrides={
+                        'load_ssh_config': False,
+                        'UserKnownHostsFile': self.known_hosts_file})
+        self.connection = Connection(
+            self.hostname,
+            self.user,
+            port=self.port,
+            config=config,
+            connect_timeout=self.connect_timeout,
+            connect_kwargs={
+                'key_filename': os.path.expanduser(self.key_file),
+            })
+        self._check_install_key_required()
+
+    def ssh_debug_cmd(self):
+        self.log.info("{0.key_file} {0.hostname} {0.user}".format(self))
+
+    def run(self, cmd, *args, **kwargs):
+        self.log.info('Running command {}'.format(cmd))
+        self.log.info("args {}, {}".format(args, kwargs))
+        result = self.connection.run(cmd, warn=True, encoding='utf-8')
+        setattr(result, 'exit_status', result.exited)
+        self.log.info('Command {} finished with result {}'.format(result.command, result.exited))
+        self.log.info('STDOUT: {}'.format(result.stdout.encode('utf-8')))
+        self.log.info('STDERR: {}'.format(result.stderr.encode('utf-8')))
+        return result
+
+    def is_up(self, *args, **kwargs):
+        cmd = 'true'
+        self.log.info(cmd)
+        try:
+            result = self.run(cmd, *args, **kwargs)
+            if result.ok:
+                return True
+        except (ssh_exception.SSHException, ssh_exception.NoValidConnectionsError) as details:
+            self.log.info(details)
+            return False
+
+    def send_files(self, *args, **kwargs):
+        self.log.info('{}; {}'.format(args, kwargs))
+        src = kwargs.get('src', '')
+        if args and not src:
+            src = args[0]
+        dst = kwargs.get('dst', '')
+        if args and not dst:
+            dst = args[1]
+
+        delete_dst = kwargs.get('delete_dst', False)
+        if delete_dst:
+            self.connection.run('rm -f {}'.dst, warn=True)
+
+        result = self.connection.put(local=src, remote=dst)
+        self.log.info('Local {} to {}'.format(result.orig_local, result.orig_remote))
+
+    def receive_files(self, *args, **kwargs):
```

Comment:
we should use `src` and `dest` as function arguments

Reply:
agree. will be updated